codegen/tests/simple/servicehandler.go

```diff
@@ -35,10 +42,32 @@ func NewServiceHandler(genCallback GenCallback, serviceInterface *ServiceInterfa
 	return &ServiceHandler{genCallback, serviceInterface}
 }
 
```

Comment:
As the current transformation does not support switch statement, use ifelse statement instead, the below code equivalent to `	serverError := common.CreateError(ctx, kind, message, cause)
	httpError := s.genCallback.MapError(ctx, serverError)
	if httpError == nil {
		switch t := cause.(type) {
                 case CustomError:
                      e := t.HTTPError()
                      httpError = &e
                 default:
                      e := common.HandleError(ctx, serverError)
                      httpError = &e
                }
	}
	httpError.WriteError(ctx, w)`